### PR TITLE
Add documentation for debugger_visualizer attribute

### PR DESCRIPTION
**Pull request by gibbyfree**
This PR adds documentation for the `debugger_visualizer` attribute ([Tracking issue for debugger_visualizer rust#95939](https://github.com/rust-lang/rust/issues/95939)). These docs describe use of the attribute with NatVis ([Add a new Rust attribute to support embedding debugger visualizers rust#91779](https://github.com/rust-lang/rust/pull/91779)) and pretty printers ([Add support for embedding pretty printers via #[debugger_visualizer] attribute rust#97028](https://github.com/rust-lang/rust/pull/97028)).

### CODE REVIEW

File: src/attributes/debugger.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
+# Debugger attributes
+
+he following [attributes] are used for enhancing the debugging experience when using third-party debuggers like GDB or LLDB.
```
Comment (dpaoliello) [inline src/attributes/debugger.md:3]: Typo

File: src/attributes/debugger.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+## The `debugger_visualizer` attribute
+
+The `debugger_visualizer` attribute can be used to embed a debugger visualizer file into the debug information generated by `rustc`.
+This enables an improved debugger experience for types outside of Rust's standard library.
+
+### Using `debugger_visualizer` with Natvis
+
+Natvis is an XML-based framework for Microsoft debuggers (such as Visual Studio and WinDbg that uses declarative rules to customize the display of types.
```
Comment (dpaoliello) [inline src/attributes/debugger.md:12]: Typo: missing end paren

File: src/attributes/debugger.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+GDB supports the use of a structured Python script, called a *pretty printer*, that describes how a type should be visualized in the debugger view.
+These scripts are embedded using the `gdb_script_file` meta item.
+For detailed information on pretty printers, refer to GDB's [pretty print documentation].
+
+Embedded pretty printers are not automatically loaded when debugging a binary under GDB.
+There are two ways to enable auto-loading embedded pretty printers:
+1. Launch GDB with extra arguments to explicitly add a directory or binary to the auto-load safe path: `gdb -iex "set auto-load safe-path path/to/binary" path/to/binary` (For more information, see GDB's [auto-loading documentation])
+1. Create a file named `gdbinit` under `$HOME/.config/gdb` (you may need to create the directory if it doesn't already exist). Add the following line to that file: `add-auto-load-safe-path path/to/binary`.
+
+Consider a crate called `foobar` with this directory structure:
```
Comment (dpaoliello) [inline src/attributes/debugger.md:93]: Please don't use `foobar`

File: src/attributes/debugger.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    pub struct Person {
+        pub name: String,
+        pub age: i32,
+    }
+}
+
+use person::Person;
+
+fn main() {
+    let bob = Person::new(String::from("Bob"), 10);
```
Comment (madsmtm) [inline src/attributes/debugger.md:116]: This example doesn't compile, which is a bit annoying when trying out the feature

Comment (gibbyfree) [reply src/attributes/debugger.md:132]: Good point - I just resolved the compile errors in both code snippets.

**Comment by ehuss** [comment]
I applied some updates to the text. A rough overview of the changes:

- General edits to maintain the correct style for the reference and consistency.
- Added more specifics.
- Removed mention of LLDB since it doesn't support LLDB AFAIK.
- Removed mentions of Cargo and `rustc`. The reference tries to focus on specifically the Rust language.
- Simplified the examples to focus on the bare minimum.
- Fixed some errors in the examples (like the natvis type path was missing the module).
- Added a print statement that one can break on. AFAIK you can't break on `()`.
- Switched suggestion of `set auto-load safe-path` to `add-auto-load-safe-path`, since the former seems to break rust-gdb.

**Review by ehuss — APPROVED** [review]

**Pull request merged**